scripts/us_census/pep/us_pep_sex/process.py

```diff
@@ -424,60 +571,65 @@ def _state_2010_2020(file_path: str) -> pd.DataFrame:
-            12: '2019',
-            13: 'April2020',
-            14: '2020'
-        }
-    })
-    df = df.rename(
-        columns={
-            'POPEST_MALE': 'Count_Person_Male',
-            'POPEST_FEM': 'Count_Person_Female',
-            'YEAR': 'Year'
+    try:
+        df = pd.read_csv(file_path, thousands=',')
+        df['geo_ID'] = 'geoId/' + (df['STATE'].map(str)).str.zfill(2)
+        df = df.replace({
+            'YEAR': {
+                1: 'April2010Census',
+                2: 'April2010Estimate',
+                3: '2010',
+                4: '2011',
+                5: '2012',
+                6: '2013',
+                7: '2014',
+                8: '2015',
+                9: '2016',
+                10: '2017',
+                11: '2018',
+                12: '2019',
+                13: 'April2020',
+                14: '2020'
+            }
         })
-    df = df.drop(columns=[
-        'SUMLEV', 'STATE', 'STNAME', 'POPESTIMATE', 'UNDER5_TOT', 'UNDER5_MALE',
-        'UNDER5_FEM', 'AGE513_TOT', 'AGE513_MALE', 'AGE513_FEM', 'AGE1417_TOT',
-        'AGE1417_MALE', 'AGE1417_FEM', 'AGE1824_TOT', 'AGE1824_MALE',
-        'AGE1824_FEM', 'AGE16PLUS_TOT', 'AGE16PLUS_MALE', 'AGE16PLUS_FEM',
-        'AGE18PLUS_TOT', 'AGE18PLUS_MALE', 'AGE18PLUS_FEM', 'AGE1544_TOT',
-        'AGE1544_MALE', 'AGE1544_FEM', 'AGE2544_TOT', 'AGE2544_MALE',
-        'AGE2544_FEM', 'AGE4564_TOT', 'AGE4564_MALE', 'AGE4564_FEM',
-        'AGE65PLUS_TOT', 'AGE65PLUS_MALE', 'AGE65PLUS_FEM', 'AGE04_TOT',
-        'AGE04_MALE', 'AGE04_FEM', 'AGE59_TOT', 'AGE59_MALE', 'AGE59_FEM',
-        'AGE1014_TOT', 'AGE1014_MALE', 'AGE1014_FEM', 'AGE1519_TOT',
-        'AGE1519_MALE', 'AGE1519_FEM', 'AGE2024_TOT', 'AGE2024_MALE',
-        'AGE2024_FEM', 'AGE2529_TOT', 'AGE2529_MALE', 'AGE2529_FEM',
-        'AGE3034_TOT', 'AGE3034_MALE', 'AGE3034_FEM', 'AGE3539_TOT',
-        'AGE3539_MALE', 'AGE3539_FEM', 'AGE4044_TOT', 'AGE4044_MALE',
-        'AGE4044_FEM', 'AGE4549_TOT', 'AGE4549_MALE', 'AGE4549_FEM',
-        'AGE5054_TOT', 'AGE5054_MALE', 'AGE5054_FEM', 'AGE5559_TOT',
-        'AGE5559_MALE', 'AGE5559_FEM', 'AGE6064_TOT', 'AGE6064_MALE',
-        'AGE6064_FEM', 'AGE6569_TOT', 'AGE6569_MALE', 'AGE6569_FEM',
-        'AGE7074_TOT', 'AGE7074_MALE', 'AGE7074_FEM', 'AGE7579_TOT',
-        'AGE7579_MALE', 'AGE7579_FEM', 'AGE8084_TOT', 'AGE8084_MALE',
-        'AGE8084_FEM', 'AGE85PLUS_TOT', 'AGE85PLUS_MALE', 'AGE85PLUS_FEM',
-        'MEDIAN_AGE_TOT', 'MEDIAN_AGE_MALE', 'MEDIAN_AGE_FEM'
-    ])
-    df = df[(df['Year'] != 'April2010Census') &
-            (df['Year'] != 'April2010Estimate') & (df['Year'] != 'April2020')]
-    df['Measurement_Method'] = 'dcAggregate/CensusPEPSurvey_PartialAggregate'
-    return df
+        df = df.rename(
+            columns={
+                'POPEST_MALE': 'Count_Person_Male',
+                'POPEST_FEM': 'Count_Person_Female',
+                'YEAR': 'Year'
+            })
+        df = df.drop(columns=[
```

Comment:
may be easier to to do df.drop(columns=df.columns.difference([<list of columns to keep>])..)

Reply:
Modified